src/background/extension.ts

```diff
@@ -336,6 +336,14 @@ export class Extension {
         setShortcut(command, shortcut);
     }
 
+    private static async isAllowedFileSchemeAccess() {
+        try {
+            return await chrome.extension.isAllowedFileSchemeAccess();
+        } catch (e) {
+            return new Promise<boolean>((resolve) => chrome.extension.isAllowedFileSchemeAccess(resolve));
```

Comment:
I cannot find documentation of why this function should accept a resolve object. Is this due some browser compatibility?

Reply:
It's just a bit of syntactic sugar: `chrome.extension.isAllowedFileSchemeAccess()` can be used with callbacks and on some browser versions with promises. This whole function just wraps callback-style call in a Promise to make `collectData()` easier to read.